### PR TITLE
Logic/Model/Storage/UI: Remark Command

### PR DESCRIPTION
**Pull request by hanselblack**
Added new command as remark:
- This is used for adding a remark on a person such as "Likes to drink coffee"

**Slight change to existing UI: PersonPanel.fxml**
- Added a new label for remark

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                personToEdit.getAddress(), remark, personToEdit.getTags());
+
+        try {
+            model.updatePerson(personToEdit, editedPerson);
+        } catch (DuplicatePersonException dpe) {
+            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+        } catch (PersonNotFoundException pnfe) {
+            throw new AssertionError("The target person cannot be missing");
+        }
+        //model.updateFilteredListToShowAll();
```
Comment (goweiwen) [inline src/main/java/seedu/address/logic/commands/RemarkCommand.java]: Can remove this line.

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            return false;
+        }
+
+        //state check
+        RemarkCommand e = (RemarkCommand) other;
+        return index.equals(e.index) && remark.equals(e.remark);
+    }
+    @Override
+    public String getCommandWord() {
+        return null;
```
Comment (goweiwen) [inline src/main/java/seedu/address/logic/commands/RemarkCommand.java]: Should return `this.COMMAND_WORD`.

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        //instanceof handles null
+        if (!(other instanceof RemarkCommand)) {
+            return false;
+        }
+
+        //state check
+        RemarkCommand e = (RemarkCommand) other;
+        return index.equals(e.index) && remark.equals(e.remark);
+    }
+    @Override
```
Comment (goweiwen) [inline src/main/java/seedu/address/logic/commands/RemarkCommand.java]: Add a blank line above the `@Override` annotation.

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 4]
```diff
@@ -11,5 +11,5 @@
     public static final Prefix PREFIX_EMAIL = new Prefix("e/");
     public static final Prefix PREFIX_ADDRESS = new Prefix("a/");
     public static final Prefix PREFIX_TAG = new Prefix("t/");
-
+    public static final Prefix PREFIX_REMARK = new Prefix("r/");
```
Comment (goweiwen) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:14]: Should we allow users to directly add people with remarks or edit remarks with the edit command if we have this prefix?

File: src/main/java/seedu/address/logic/parser/RemarkCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+        Index index;
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (IllegalValueException ive) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RemarkCommand.MESSAGE_USAGE));
+        }
+
+        String remark = argMultimap.getValue(PREFIX_REMARK).orElse("");
```
Comment (goweiwen) [inline src/main/java/seedu/address/logic/parser/RemarkCommandParser.java:36]: Should make use of `arePrefixesPresent()` method as defined below.

**Review by goweiwen — COMMENTED** [review]
Nice! I added a few comments that you might want to fix before merging.

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java (hunk 1 above)
Comment (hanselblack) [reply src/main/java/seedu/address/logic/commands/RemarkCommand.java]: Noted. Thanks, I have removed that line.

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java (hunk 3 above)
Comment (hanselblack) [reply src/main/java/seedu/address/logic/commands/RemarkCommand.java]: Noted. Thanks, added a blank line.

File: src/main/java/seedu/address/logic/parser/RemarkCommandParser.java (hunk 5 above)
Comment (hanselblack) [reply src/main/java/seedu/address/logic/parser/RemarkCommandParser.java:36]: Noted. Removed the unused method defined below.

**Pull request merged**